src/pages/BookDetails/BookDetails.js

```diff
@@ -1,80 +1,116 @@
 import * as React from 'react';
-import Box from '@mui/material/Box';
-import Grid from '@mui/material/Grid';
-import Typography from "@mui/material/Typography";
-import {Chip, Link, ListItem, Paper, Rating} from "@mui/material";
 import {useSelector, useDispatch} from "react-redux";
 import {useParams} from "react-router-dom";
 import {useEffect, useState} from "react";
-import {fetchBook} from "store/books/booksActions";
+import {Chip, Fab, Link, ListItem, Paper, Rating, Stack} from "@mui/material";
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import Typography from "@mui/material/Typography";
+import List from "@mui/material/List";
+import Divider from "@mui/material/Divider";
 import StarIcon from "@mui/icons-material/Star";
+import AddIcon from "@mui/icons-material/Add";
+import FavoriteIcon from "@mui/icons-material/Favorite";
+import {fetchBook} from "store/books/booksActions";
 import api from "api/api";
 import Review from "components/Review";
-import List from "@mui/material/List";
-import Divider from "@mui/material/Divider";
+import NewReview from "components/NewReview";
 import classes from "./BookDetails.module.scss";
 
 
 const BookDetails = () => {
     let {id} = useParams();
     const dispatch = useDispatch();
     const currentBook = useSelector((state) => state.booksReducer.currentBook);
-    const {title, author, description, photo, rating, genres} = currentBook;
+    const user = useSelector((state) => state.authReducer.user);
+    const {title, author, description, photo, rating, genres, numberOfReviews} = currentBook;
     const [reviews, setReviews] = useState([]);
+    const [newReviewForm, setNewReviewForm] = useState(false);
+    const [favorite, setFavorite] = useState(false);
+
+    const discardHandler = () => setNewReviewForm(false);
 
+    const getReviews = () => {
+        api.getReviews(+id).then(res => setReviews(res));
+        setNewReviewForm(false);
+    }
+
+    const favoriteHandler = () => {
+        if (!favorite) {
+            api.addBookToCollection(+id, title, author, genres, rating, numberOfReviews, photo, description, user.username).then(res => setFavorite(true));
+        }
+        else api.removeBookFromCollection(user.username, +id).then(res => setFavorite(false));
```

Comment:
```suggestion
        else { 
           api.removeBookFromCollection(user.username, +id).then(res => setFavorite(false));
         }
```
i noticed throughout the code you use inline if - else. We generally avoid that since it can easily lead to bugs: you want to add another line to `else` but forget to wrap it in `{}` and then the line executes every time. It's quite common.